packages/spec-parser/src/specParser.browser.ts

```diff
@@ -163,6 +165,21 @@ export class SpecParser {
     throw new Error("Method not implemented.");
   }
 
+  /**
+   * Generates and update artifacts from the OpenAPI specification file. Generate Adaptive Cards, update Teams app manifest, and generate a new OpenAPI specification file.
+   * @param manifestPath A file path of the Teams app manifest file to update.
+   * @param filter An array of strings that represent the filters to apply when generating the artifacts. If filter is empty, it would process nothing.
+   * @param outputSpecPath File path of the new OpenAPI specification file to generate. If not specified or empty, no spec file will be generated.
+   */
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async generateForCopilot(
+    manifestPath: string,
+    filter: string[],
+    outputSpecPath: string,
+    signal?: AbortSignal
+  ): Promise<GenerateResult> {
```

Comment:
should we also make ai-plugin.json as an input parameter? I think user may customize the file location when they want to use the codelens to add a new API?

Reply:
Sure